pages/tutorials/3.0/graphics/vertex-array.md

```diff
@@ -460,11 +460,15 @@ private:
 
     void resetParticle(std::size_t index)
     {
+        // create random number generator
+        static std::random_device rd;
+        static std::mt19937       rng(rd());
+
         // give a random velocity and lifetime to the particle
-        const sf::Angle angle       = sf::degrees(std::rand() % 360);
-        const float     speed       = (std::rand() % 50) + 50.f;
+        const sf::Angle angle       = sf::degrees(std::uniform_real_distribution(0.f, 360.f)(rng));
+        const float     speed       = std::uniform_real_distribution(50.f, 100.f)(rng);
         m_particles[index].velocity = sf::Vector2f(speed, angle);
-        m_particles[index].lifetime = sf::milliseconds((std::rand() % 2000) + 1000);
+        m_particles[index].lifetime = sf::milliseconds(std::uniform_int_distribution(1000, 3000)(rng));
```

Comment:
If we use the distribution multiple times, shouldn't it be reused?

Reply:
yes to that as well. The example on cppreference shows is quite good.

Reply:
This needs to be considered from a design point of view. Should the user have to call something like sf::init_random(), or should this function check that every single time and initialize it if needed.

Reply:
The question is for the example should an init_random function be created, or should the reset_particle function check that it is initialized and do it if it isn't. Then to the example main the call to init random would need to be added

Reply:
> If we use the distribution multiple times, shouldn't it be reused?

I'm not aware of any benefit of doing so, so I chose to keep the code as simple as possible. Perhaps there are marginal performance implications but I think such considerations are beyond the scope of a simple snippet of example code.

Reply:
Tested it, and indeed there is no difference at least on libc++ and libstdc++. This is a missed optimization opportunity in the implementation from what I can see. https://quick-bench.com/q/KLaNdK7CiLyC-nQPg1YVJ5t7CeQ

Reply:
However local google benchmark run with clang 19 using MSVC STL suggests something else:
```
Run on (22 X 2995 MHz CPU s)
CPU Caches:
  L1 Data 48 KiB (x11)
  L1 Instruction 64 KiB (x11)
  L2 Unified 2048 KiB (x11)
  L3 Unified 24576 KiB (x1)
---------------------------------------------------------
Benchmark               Time             CPU   Iterations
---------------------------------------------------------
OneDistInt         144355 ns       122768 ns         5600
ManyDistsInt       176374 ns       142997 ns         4480
OneDistFloat        99271 ns        81961 ns         8960
ManyDistsFloat     138700 ns       112997 ns         4978
```

Reply:
While I personally would in my own code reuse the distribution, I think it's okay for the given example code.